### PR TITLE
TW-218: Fix double click on forward msg

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #218 

- Resolved: 

https://github.com/linagora/twake-on-matrix/assets/99852347/3fe1f03a-082d-45e3-b125-54892b3f9d5a

### CODE REVIEW

File: lib/config/routes.dart [hunk 1]
```diff
@@ -348,32 +356,38 @@ class AppRoutes {
...
+    VWidget(
+      path: 'emotes/:state_key',
+      widget: const EmotesSettings(),
+      buildTransition: _dynamicTransition,
+    ),
+    VWidgetWithDependency(
+      path: 'forward',
+      widget: const Forward(),
+      di: ForwardDI(),
+      buildTransition: _fadeTransition,
```
Comment (sherlockvn) [inline lib/config/routes.dart:388]: IMO, use _leftToRight transition

File: lib/domain/usecase/forward/forward_message_interactor.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        yield* _forwardMessage(message, room);
+        matrixState.shareContent?.clear();
+      }
+      yield Right(ForwardMessageSuccess(room));
+    } catch (exception) {
+      yield Left(ForwardMessageFailed(exception: exception));
+    }
+  }
+
+  Stream<Either<Failure, Success>> _forwardMoreMessageAction({
```
Comment (sherlockvn) [inline lib/domain/usecase/forward/forward_message_interactor.dart:56]: _forwardMultipleMessagesAction

File: lib/domain/usecase/forward/forward_message_interactor.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
+
+
```
Comment (imGok) [inline lib/domain/usecase/forward/forward_message_interactor.dart:2]: delete these 2 lines pls
```suggestion
```

File: lib/domain/usecase/forward/forward_message_interactor.dart [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  Stream<Either<Failure, Success>> _forwardOneMessageAction({
+    required Room room,
+    required MatrixState matrixState,
+  }) async* {
+    try {
+      final message = matrixState.shareContent;
+      if (message != null) {
+        yield* _forwardMessage(message, room);
+        matrixState.shareContent?.clear();
```
Comment (imGok) [inline lib/domain/usecase/forward/forward_message_interactor.dart:48]:
```suggestion
        message.clear();
```

File: lib/domain/usecase/forward/forward_message_interactor.dart [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      final messages = matrixState.shareContentList;
+      if (messages.isNotEmpty) {
+        for (final message in messages) {
+          if (message != null) {
+            yield* _forwardMessage(message, room);
+          } else {
+            continue;
+          }
+        }
+        matrixState.shareContentList.clear();
```
Comment (imGok) [inline lib/domain/usecase/forward/forward_message_interactor.dart:72]:
```suggestion
        messages.clear();
```

File: lib/pages/forward/forward_view.dart [hunk 6]
```diff
@@ -53,24 +57,49 @@ class _ForwardViewState extends State<ForwardView> {
...
+              (success) {
+                if (success is ForwardMessageLoading) {
+                  return SizedBox(
+                    height: ForwardViewStyle.bottomBarHeight,
+                    child: const Align(
+                      alignment: Alignment.centerRight,
+                      child: Padding(
+                        padding: EdgeInsets.only(right: 14),
+                        child: TwakeFloatingActionButton(
+                            customIcon: SizedBox(child: CircularProgressIndicator())
```
Comment (imGok) [inline lib/pages/forward/forward_view.dart:77]:
```suggestion
                          customIcon: SizedBox(child: CircularProgressIndicator())
```

File: lib/widgets/twake_components/twake_fab.dart [hunk 7]
```diff
@@ -46,15 +49,17 @@ class TwakeFloatingActionButton extends StatelessWidget {
...
-                  : null,
+                icon,
+                size: size,
+                color: Theme.of(context).colorScheme.onPrimaryContainer)
+              : customIcon != null
+                ? SizedBox(
+                  width: 24.0,
+                  height: 24.0,
+                  child: customIcon)
+                : imagePath != null ? SvgPicture.asset(imagePath!) : null,
```
Comment (imGok) [inline lib/widgets/twake_components/twake_fab.dart:62]: IMO: 3 nested ternaries is hard to read

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**